### PR TITLE
doc: improve doc for add user to GKE Kubeflow

### PR DESCRIPTION
**Pull request by Bobgy**
This should answer future questions like https://kubeflow.slack.com/archives/C7REE0EHK/p1597386829039400?thread_ts=1597294388.025200&cid=C7REE0EHK.

Also, updated next steps for deploy-cli doc.

/assign @jlewi @joeliedtke

### CODE REVIEW

**Comment by joeliedtke** [comment]
Preview:
https://deploy-preview-2133--competent-brattain-de2d6d.netlify.app/docs/components/multi-tenancy/
https://deploy-preview-2133--competent-brattain-de2d6d.netlify.app/docs/gke/customizing-gke/
https://deploy-preview-2133--competent-brattain-de2d6d.netlify.app/docs/gke/deploy/delete-cli/

File: content/en/docs/gke/deploy/deploy-cli.md [hunk 1]
```diff
@@ -203,6 +203,16 @@ Follow these steps to verify the deployment:
 
 Follow these steps to access the Kubeflow central dashboard:
 
+1. You need the
```
Comment (joeliedtke) [inline content/en/docs/gke/deploy/deploy-cli.md:206]: Suggested revision:

Use the following command to grant yourself the [IAP-secured Web App User](https://cloud.google.com/iap/docs/managing-access) role:

File: content/en/docs/gke/deploy/deploy-cli.md [hunk 2]
```diff
@@ -203,6 +203,16 @@ Follow these steps to verify the deployment:
...
+  [IAP-secured Web App User](https://cloud.google.com/iap/docs/managing-access)
+  role:
+
+    ```
+    gcloud projects add-iam-policy-binding [PROJECT] --member=user:[EMAIL] --role=roles/iap.httpsResourceAccessor
+    ```
+
+    Note, you need the `IAP-secured Web App User` role even if you are already an owner or editor of the project. `IAP-secured Web App User` role is not implied by just `Project Owner` or `Project Editor` roles.
```
Comment (joeliedtke) [inline content/en/docs/gke/deploy/deploy-cli.md:214]: just -> the

File: content/en/docs/gke/customizing-gke.md [hunk 3]
```diff
@@ -146,6 +146,42 @@ For example, to modify settings for the Jupyter web app:
...
+
+* To access the Kubeflow UI through IAP, the user needs the
+  [IAP-secured Web App User](https://cloud.google.com/iap/docs/managing-access)
+  role:
+
+    ```
+    gcloud projects add-iam-policy-binding [PROJECT] --member=user:[EMAIL] --role=roles/iap.httpsResourceAccessor
+    ```
+
+    Note, you need to grant the user `IAP-secured Web App User` role even if the user is already an owner or editor of the project. `IAP-secured Web App User` role is not implied by just `Project Owner` or `Project Editor` roles.
```
Comment (joeliedtke) [inline content/en/docs/gke/customizing-gke.md:174]: just -> the

File: content/en/docs/gke/customizing-gke.md [hunk 4]
```diff
@@ -146,6 +146,42 @@ For example, to modify settings for the Jupyter web app:
...
+
+* To be able to run `gcloud get credentials` and see logs in Cloud Logging
```
Comment (joeliedtke) [inline content/en/docs/gke/customizing-gke.md:176]: Should `gcloud get credentials` be `gcloud container clusters get-credentials`?

File: content/en/docs/gke/deploy/deploy-cli.md [hunk 5]
```diff
@@ -327,10 +337,8 @@ The service accounts are:
   example](https://github.com/kubeflow/examples/tree/master/github_issue_summarization/pipelines).
 * See how to [delete](/docs/gke/deploy/delete-cli/) your Kubeflow deployment 
   using the CLI.
+* See how to [add users to Kubeflow](/docs/gke/customizing-gke/#add-users-to-kubeflow).
```
Comment (8bitmp3) [inline content/en/docs/gke/deploy/deploy-cli.md:340]: Suggestion:

- For a11y: "~~see~~" -> rephrase or use "go to", if you can.

```diff
* Run a full ML workflow on Kubeflow ..
..
- * See how to [delete](/docs/gke/deploy/delete-cli/) your Kubeflow deployment 
- using the CLI.
+ * Learn how to
+ [delete your Kubeflow deployment using the CLI](/docs/gke/deploy/delete-cli/).
- * See how to [add users to Kubeflow](/docs/gke/customizing-gke/#add-users-to-kubeflow).
+ * To add users to Kubeflow, go to
+ [a dedicated section in Customizing Kubeflow on GKE](/docs/gke/customizing-gke/#add-users-to-kubeflow).
- * See how to [customize](/docs/gke/customizing-gke/) your Kubeflow
-   deployment.
+ * To taylor your GKE deployment on Kubeflow, go to
+ [Customizing Kubeflow on GKE](/docs/gke/customizing-gke/).
```

File: content/en/docs/gke/deploy/deploy-cli.md [hunk 6]
```diff
@@ -327,10 +337,8 @@ The service accounts are:
...
 * See how to [customize](/docs/gke/customizing-gke/) your Kubeflow 
   deployment.
-* See how to [upgrade Kubeflow](/docs/upgrading/upgrade/) and how to 
-  [upgrade or reinstall a Kubeflow Pipelines 
-  deployment](/docs/pipelines/upgrade/).
 * [Troubleshoot](/docs/gke/troubleshooting-gke/) any issues you may
```
Comment (8bitmp3) [inline content/en/docs/gke/deploy/deploy-cli.md:343]: Proposal:

```diff
- * [Troubleshoot](/docs/gke/troubleshooting-gke/) any issues you may
- find.
+ * For troubleshooting Kubeflow deployments on GKE, go to
+ the [Troubleshooting deployments](/docs/gke/troubleshooting-gke/) guide.
```

File: content/en/docs/components/multi-tenancy/getting-started.md [hunk 7]
```diff
@@ -101,6 +97,8 @@ the user's email address:
     gcloud projects add-iam-policy-binding [PROJECT] --member=user:[EMAIL] --role=roles/iap.httpsResourceAccessor
     ```
 
+    Note, you need to grant the user `IAP-secured Web App User` role even if the user is already an owner or editor of the project. `IAP-secured Web App User` role is not implied by just `Project Owner` or `Project Editor` roles.
```
Comment (8bitmp3) [inline content/en/docs/components/multi-tenancy/getting-started.md:100]: Suggestion:

Correct me if I'm wrong, for better UX you could bolden **Note** here and elsewhere, if you see fit.

e.g. 

```
**Note:*** You need to grant...
```

File: content/en/docs/components/multi-tenancy/getting-started.md [hunk 8]
```diff
@@ -77,15 +73,15 @@ deployed Kubeflow will have administration privileges in the cluster.
 
 ### Pre-requisites: grant user minimal Kubernetes cluster access
 
-You must grant each user the minimal permission scope that allows them to 
+You must grant each user the minimal permission scope that allows them to
 connect to the Kubernetes cluster.
 
-For example, for Google Cloud Platform (GCP) users, you should grant the 
+For example, for Google Cloud users, you should grant the
```
Comment (8bitmp3) [inline content/en/docs/components/multi-tenancy/getting-started.md:79]: Here, it's `Google Cloud` with `Platform (GCP)` removed. I suggest that for consistency you could apply this in other files in this PR. I've highlighted a few of them for you to make it easier.

File: content/en/docs/components/multi-tenancy/getting-started.md [hunk 9]
```diff
@@ -115,7 +113,7 @@ Kubeflow {{% kf-latest-version %}} provides automatic profile creation:
   - The Kubeflow deployment process automatically creates a profile for the user
     performing the deployment. When the user access the Kubeflow central dashboard
     they see their profile in the dropdown list.
-  - The automatic profile creation can be disabled as part of the deployment by setting the registration-flow env variable to false. And an admin can manually create profiles per user or per project and add collaborators through YAML files. 
+  - The automatic profile creation can be disabled as part of the deployment by setting the registration-flow env variable to false. And an admin can manually create profiles per user or per project and add collaborators through YAML files.
    Modify the kustomize/centraldashboard/base/parama.env to set the registration variable to false
```
Comment (8bitmp3) [inline content/en/docs/components/multi-tenancy/getting-started.md:117]: Suggestion:
- Don't want to be _that user_ but adding a blank line before and after code samples would be good IMHO.

File: content/en/docs/components/multi-tenancy/getting-started.md [hunk 10]
```diff
@@ -347,7 +345,7 @@ Run the following command to create the corresponding contributor resources:
 ```
 kubectl create -f rolebinding.yaml
```
Comment (8bitmp3) [inline content/en/docs/components/multi-tenancy/getting-started.md:347]: This extra line might not be necessary, for consistency.

File: content/en/docs/gke/customizing-gke.md [hunk 11]
```diff
@@ -230,16 +266,6 @@ More detailed instructions follow.
       * one for GPU machines, in [`cluster.jinja`](https://github.com/kubeflow/manifests/tree/{{< params "githubbranch" >}}/gcp/deployment_manager_configs/cluster.jinja#L140).
```
Comment (8bitmp3) [inline content/en/docs/gke/customizing-gke.md:266]: Line 228 (sorry, I couldn't select it)

- For consistency: `Google Cloud` ~~`Platform`~~/~~`(GCP)`~~, as you did in the other file.

File: content/en/docs/gke/customizing-gke.md [hunk 12]
```diff
@@ -146,6 +146,42 @@ For example, to modify settings for the Jupyter web app:
...
+  (formerly Stackdriver), the user needs viewer access on the project:
+
+    ```
+    gcloud projects add-iam-policy-binding [PROJECT] --member=user:[EMAIL] --role=roles/viewer
+    ```
+
+Alternatively, you can also grant these roles on the [IAM page in the Cloud Console](https://console.cloud.google.com/iam-admin/iam). Make sure you are in the same project as your Kubeflow deployment.
+
 <a id="gpu-config"></a>
 ### Add GPU nodes to your cluster
```
Comment (8bitmp3) [inline content/en/docs/gke/customizing-gke.md:187]: Line 191 (sorry, I couldn't select it)

- For consistency: `Google Cloud` ~~`Platform`~~/~~`(GCP)`~~, as you did in the other file.

File: content/en/docs/gke/deploy/deploy-cli.md [hunk 13]
```diff
@@ -327,10 +337,8 @@ The service accounts are:
   example](https://github.com/kubeflow/examples/tree/master/github_issue_summarization/pipelines).
```
Comment (8bitmp3) [inline content/en/docs/gke/deploy/deploy-cli.md:337]: Lines 3, 8, 19, 27, 295, 298, 304, 317 and 325 (sorry, I couldn't comment on them directly):

- For consistency: `Google Cloud` ~~`Platform`~~/~~`(GCP)`~~, as you did in the other file.

File: content/en/docs/components/multi-tenancy/getting-started.md [hunk 14]
```diff
@@ -174,7 +172,7 @@ The following resources are created as part of the profile creation:
   - A Kubernetes namespace that shares the same name with the corresponding
     profile.
   - Kubernetes RBAC role binding for the namespace: *Admin*. This makes the
```
Comment (8bitmp3) [inline content/en/docs/components/multi-tenancy/getting-started.md:174]: Suggestion:

- a11y: To help new users, maybe we can spell out certain abbreviations/terminology, even if it's common knowledge (Kubernetes has so many interesting terms, it needs its own dictionary 🙂)

```diff
-  - Kubernetes RBAC role binding for the namespace: *Admin*. This makes the
+ Kubernetes RBAC ([Role-based access control](https://kubernetes.io/docs/reference/access-authn-authz/rbac/))
+ role binding for the namespace: *Admin*. This makes the
...
```

File: content/en/docs/gke/deploy/deploy-cli.md (hunk 1 above)
Comment (8bitmp3) [reply content/en/docs/gke/deploy/deploy-cli.md:206]: I couldn't commend on line 172:

```diff
- To deploy kubeflow just run
+ To deploy Kubeflow, run the following command:

[triple back tick]
make apply
[triple back tick ]
``` 

I would add also colons (":") where you see them missing before each command, but it's probably not a priority here.

**Comment by 8bitmp3** [comment]
Hi @Bobgy, since you're going through the GKE docs, do you think this should be included in this PR or a new one? https://github.com/kubeflow/website/issues/1997#issue-638027038 (the link that should potentially be updated is: https://www.kubeflow.org/docs/gke/troubleshooting-gke/#gke-certificate-fails-to-be-provisioned)

> "In troubleshooting instructions for deleting the GKE Certificate, the listed command is `kubectl -n istio-system delete ingress envoy-ingress`, rather than `kubectl -n istio-system delete managedcertificate gke-certificate`"

I'd be happy to assist if you need an extra pair of hands. Cheers!

File: content/en/docs/components/multi-tenancy/getting-started.md (hunk 7 above)
Comment (Bobgy) [reply content/en/docs/components/multi-tenancy/getting-started.md:100]: That's a great suggestion!

File: content/en/docs/components/multi-tenancy/getting-started.md (hunk 9 above)
Comment (Bobgy) [reply content/en/docs/components/multi-tenancy/getting-started.md:117]: Thank you! Is there some guideline on suggested format? Is there a tool that can automate the formatting?

I was previously using `prettier` for yaml formatting, but I can only use it for new pages. Reformatting an existing page with an auto formatter causes too much distraction for the review.

File: content/en/docs/components/multi-tenancy/getting-started.md (hunk 14 above)
Comment (Bobgy) [reply content/en/docs/components/multi-tenancy/getting-started.md:174]: Good point!

**Comment by Bobgy** [comment]
Thank you @joeliedtke @8bitmp3!
I've addressed your comments.

**Comment by Bobgy** [comment]
I'd want to submit this PR quickly, because some missing information "how to add user to Kubeflow" needs a quick update.

/approve

**Comment by 8bitmp3** [comment]
/lgtm

Thank you @Bobgy for the awesome update

@joeliedtke @jlewi WDYT?

**Pull request merged**